src/address_finder.rs

```diff
@@ -14,54 +14,128 @@ pub enum AddressFinderError {
     #[fail(display = "No process with PID: {}", _0)] NoSuchProcess(pid_t),
     #[fail(display = "Permission denied when reading from process {}. Try again with sudo?", _0)]
     PermissionDenied(pid_t),
+    #[fail(display = "Couldn't get port for PID {}. Possibilities: that process doesn't exist or you have SIP enabled and you're trying to profile system Ruby (try rbenv instead).", _0)]
+    MacPermissionDenied(pid_t),
```

Comment:
not 100% on this error variant's existence. at least `#[cfg(target_os = "macos")]` it.